internal/dataplane/kong_client.go

```diff
@@ -178,6 +178,10 @@ type KongClient struct {
 	// While lastProcessedSnapshotHash keeps track of the last processed cache snapshot (the one kept in KongClient.cache),
 	// lastValidCacheSnapshot can also represent the fallback cache snapshot that was successfully synced with gateways.
 	lastValidCacheSnapshot *store.CacheStores
+
+	// konnectConfigSynchronizer receives latest successfully applied Kong configuration from KongClient
+	// and uploads it to Konnect.
+	konnectConfigSynchronizer *konnect.ConfigSynchronizer
```

Comment:
Could we make this dependency an interface instead of relying on a specific type? As I understand, we only need to call `SetTargetContent` from `KongClient`. This should allow easier testing. In `KongClient` we'd only ensure an expected target was set in a mock, while we could have a separate test suite for the `konnect.ConfigSynchronizer`, testing them in separation.

If you agree, this could be done in a separate PR.

Reply:
I considered to define is as an `interface`. There is not a specific reason to use pointer to a concrete type, but I thought that we are already using many interfaces, and it is not likely to have a different implementation here so I used it. If you think using interface is better, let's do it in another PR.